doc/decisions/0_drafts/buffer_struct.md

```diff
@@ -0,0 +1,189 @@
+# Buffer Struct
+
+## Problem
```

Comment:
> **Note**: The problem section is very sparse, because there isn't really a huge problem here. This is more of a proposal than decision for a solution to a problem.

Reply:
Especially abstractions should have a use case and a clear problem. E.g. avoiding code duplication is a common use case for abstraction. If it doesn't solve a problem we definitely shouldn't do it.

Reply:
Can you please just read my proposal and give an opinion instead of wasting both our time with criticising the process.... Why do you think I added this note when I created the PR? I know this doesn't fit the decision process. But if I had created a proposal issue, you'd have complained as well...

You can always invent some problem or come up with some abstract or even hypothetical use case. I even tried to come up with something instead of just leaving the section empty. You still found a reason to ignore the proposal.

Reply:
I already read the problem several times: it is simply not convincing that anything should be done. I would concentrate our efforts on something that brings benefit for Elektra's user and not do internal restructuring just to have a more uniform API, i.e. less internal functions (which is currently the only reason given). Having a more generic internal API/less functions by itself **isn't better**, hence no problem and hence nothing to do. It would be very different if you want to improve the public API...

I already gave you several hints which problems might be worth solving with buffers.  If you cannot add any problem in the problem section, rejecting it as the obvious choice. More comments or notes will not help as this does not lead anywhere. Please don't reply unless you have **questions** how to proceed.

Please let us focus on problems we all agree exist, there are more than enough of those. We should be happy if something works well and no extra work is needed, shouldn't we?

Btw. in any case I appreciate that you wrote this, even if we reject it as there is **currently** no immanent benefit. Then we have documented how we could introduce buffers if they are needed at some time (e.g. public API is clumsy to use).

Reply:
> I already read the problem several times

**Did you read the rest?**

I know this is not the protocol for a decision. But I clearly stated that this is more of a proposal and doesn't fit the "we have a problem, let's decide a solution" process...

> It would be very different if you want to improve the public API...

**That is a goal here**

There are some examples further down in the file.... I just didn't want to design a whole public API based on a public buffer struct, only to find out, e.g. "no we don't want public structs".

> Then we have documented how we could introduce buffers if they are needed at some time (e.g. public API is clumsy to use).

Why do you think I wrote this? The current API is okay, but with #4715 anything that needs a keyname will become pretty annoying, if we need to pass a size everywhere.

> Please don't reply unless you have questions how to proceed.

Okay, I have a question: How can I convince you to be less stubborn and get you to just spend maybe five minutes reading the whole file?